metricsaggregator/src/main/java/io/dockstore/metricsaggregator/MetricsAggregatorS3Client.java

```diff
@@ -63,26 +64,50 @@ public MetricsAggregatorS3Client(String bucketName, String s3EndpointOverride) t
 
     public void aggregateMetrics(ExtendedGa4GhApi extendedGa4GhApi) {
         List<String> metricsDirectories = getDirectories();
+
+        if (metricsDirectories.isEmpty()) {
+            System.out.println("No directories found to aggregate metrics");
+            return;
+        }
+
         // Each directory contains metrics for a specific tool version and platform
         for (String directory : metricsDirectories) {
             String toolId = S3ClientHelper.getToolId(directory); // Check if we should just give the full key
             String versionName = S3ClientHelper.getVersionName(directory);
             String platform = S3ClientHelper.getPlatform(directory);
-            List<MetricsData> metricsDataList = metricsDataS3Client.getMetricsData(toolId, versionName, Partner.valueOf(platform));
-            List<Execution> executions = metricsDataList.stream().map(metricsData -> {
-                try {
-                    String fileContent = metricsDataS3Client.getMetricsDataFileContent(metricsData.toolId(), metricsData.toolVersionName(),
-                            platform, metricsData.fileName());
-                    return List.of(GSON.fromJson(fileContent, Execution[].class));
-                } catch (IOException e) {
-                    LOG.error("Error aggregating metrics: Unable to get all executions", e);
-                    throw new RuntimeException("Error aggregating metrics: Unable to get all executions");
-                }
-            }).flatMap(List::stream).toList();
-
-            getAggregatedMetrics(executions).ifPresent(
-                    metrics -> extendedGa4GhApi.aggregatedMetricsPut(metrics, platform, toolId, versionName));
+
+            List<Execution> executions;
+            try {
+                executions = getExecutions(toolId, versionName, platform);
+            } catch (Exception e) {
+                LOG.error("Error aggregating metrics: Could not get all executions from directory {}", directory, e);
```

Comment:
I think the `e` argument will get ignored; it won't get substituted into the string and it won't be logged as an exception -- it's annoying when you use the logger string substitution feature, you forfeit the exception logging.

Reply:
We use slf4j version 1.7.36, which should support logging the exception according to https://www.slf4j.org/faq.html#paramException